Added support for Zim wiki syntax

alexivkin opened the pull request:
[Zim](http://zim-wiki.org/index.html) is a popular desktop wiki that uses a fairly simple text file [syntax] (http://zim-wiki.org/manual/Help/Wiki_Syntax.html). This writer is able to convert whatever pandoc reads into what zim expects. The reader is not needed as Zim uses pandoc to export to markdown and HTML natively.

jgm commented:
Thanks.

1.  We need tests before I'd consider merging.  At minimum you should generate `tests/writer.zim`, check it carefully for correctness, and add to the "other writers" list around line 169 of  `tests/Tests/Old.hs` 

2.  On quick glance, the indentFromHTML business seemed strange -- what is this for?

alexivkin commented:
Added the tests and fixed table syntax a bit. indentFromHTML allows ordered lists to be controlled from raw html code. I.e. if the source file includes <li>, the converted list will add a new item.
This is necessary for conversion from mediawiki for example, that often drops down to straight html for ordered lists, because of the very rigid syntax.

alexivkin commented:
The builds in this branch will fail until the submodule is updated with the Zim wiki template with the pull 202 [https://github.com/jgm/pandoc-templates/pull/202]. 
I am new to submodules, so I created two separate pulls. If there is a way to do it smoother, let me know.

alexivkin commented:
The submodule was merged, but the actual zim code was not

jgm commented:
+++ Alex Ivkin [Jul 04 16 00:18 ]:
>   The submodule was merged, but the actual zim code was not

The code was all merged.  However, I rebased it first,
squashing to a single commit.  That's probably why
github didn't think it was merged.

See commits:
a73c95f61dbf47bbb54345855b07127d9fd82e62
7fdcd9a6e212dc02e0a37f47240e5978683d66a1
b3382cf3777dc121c61937e64890cc9646dc30bf

alexivkin commented:
Thank you. For the future, what is a proper way to create pull requests that include changes to submodules? Can it be squashed on my side into a single commit between two git repos?

jgm commented:
You need to submit a separate PR in the submodule
repository, just as you did.  (And indicate on your
other PR that it depends on the submodule change, as
you also did.)

So, what you did was just right.  The only thing I
changed was that I squashed several commits in pandoc
into one.

+++ Alex Ivkin [Jul 12 16 01:53 ]:
>   Thank you. For the future, what is a proper way to create pull requests
>   that include changes to submodules? Can it be squashed on my side into
>   a single commit between two git repos?
>
>   —
>   You are receiving this because you modified the open/close state.
>   Reply to this email directly, [1]view it on GitHub, or [2]mute the
>   thread.
>
>References
>
>   1. https://github.com/jgm/pandoc/pull/3005#issuecomment-231978211
>   2. https://github.com/notifications/unsubscribe/AAAL5Drig20ueTS_vlFYoJlZf0FpvDf-ks5qU1YcgaJpZM4JAzkX
